### ISSUE TITLE
Mobile Safari clean-up/coverage

### ISSUE THREAD

**Issue by jaydiablo** [browscap/browscap#1645]
I hinted at this in the comments of #1643, but right now the mobile-safari files include a lot of iOS platforms that probably shouldn't be there.

Generally, especially with the more recent versions, the Safari version number in the useragent matches the major version of the iOS version (so iOS 10.0, 10.1, 10.2, 10.3 will all have Safari version 10.0 (based on the `Version/` token in the UA)).  Most of the patterns in the mobile-safari files seem to have +-2 or more major versions (Safari 9.x file has iOS 7.0 - 10.3, for example).

Safari on iOS devices is only updated when the iOS version is updated, and it's not possible to keep an old version of Safari on the device when it's upgraded to a newer OS version.

Another thing that is suspicious is the versions of Safari that aren't X.0 (have a minor release other than 0).  For example, in the `10-0-on` file currently, there are 4 versions of 10.x Safari (10.0, 10.1, 10.2, 10.3) which does match the available versions of iOS 10.x currently, but Safari doesn't report that version in the UA (it only reports 10.0).  This is the highest hitting Safari 10.x UA on iOS that we saw this year (by quite a bit):

>Mozilla/5.0 (iPhone; CPU iPhone OS 10_3_2 like Mac OS X) AppleWebKit/603.2.4 (KHTML, like Gecko) Version/10.0 Mobile/14F89 Safari/602.1

Notice that even though the OS version is 10_3_2, the `Version` version is still `10.0`. I can confirm this with my own device as well (which is on 10.3.3 iOS version, but still shows 10.0 as the Safari version).

This is the highest hitting non-10.0 version that we saw this year:

>Mozilla/5.0 (iPhone; CPU iPhone OS 10_3_1 like Mac OS X) AppleWebKit/600.1.4 (KHTML, like Gecko) Version/10.3.1 Mobile/14E304 Safari/600.1.4

Relative to the other UA, we saw this one **.06%** as much.  If I look at all 10.0 useragents and compare quantity to all 10.x (that aren't 10.0), we only see the 10.1/2/3 useragents **0.03%** as much, they're very rare, relatively.  I suspect they're most likely spoofed.

I'm planning to dig up a lot of mobile Safari UAs and analyze them a bit, excluding any that are obvious spoofs, then use the remaining to determine what versions of iOS should stay.

I'll post more details of the analysis as I get there.

**Comment by jaydiablo** [comment]
The first batch that I took a look at are for Safari 3.0 and 3.1.  I gathered as many 3.x useragents as I could find, then removed the obvious spoofs, and then took a look at the "Safari" version (should be one of the values here: https://en.wikipedia.org/wiki/Safari_version_history#iOS), and then I also looked at the "Mobile" build number, which should correspond to an iOS build number (which should also match the iOS version in the platform section, if provided).

From this, it appears that Safari 3.x was only available for iOS 1.x and 2.x.  3.0 was only released on iOS 1.x (and doesn't have the iOS version in the UA, but can be determined from the Mobile build number). 3.1 was only available for iOS 2.x (and does have the iOS version in the UA).

The iPad wasn't out yet, so there shouldn't ever be Safari 3.x on iPad.

I'm planning to make changes to the 3.x files in response to this. Mainly it will be splitting 3.0 and 3.1 to separate files and removing all of the unnecessary iOS versions (and iPad).

Here are a few sample "clean" 3.x UAs (where the build number matches iOS version, safari version is acceptable, format looks good):

```
Mozilla/5.0 (iPhone; U; CPU like Mac OS X; en) AppleWebKit/420.1 (KHTML, like Gecko) Version/3.0 Mobile/3A110a Safari/419.3
Mozilla/5.0 (iPhone; U; CPU like Mac OS X; en) AppleWebKit/420+ (KHTML, like Gecko) Version/3.0 Mobile/1C28 Safari/419.3
Mozilla/5.0 (iPod; U; CPU iPhone OS 2_2_1 like Mac OS X; ru-ru) AppleWebKit/525.18.1 (KHTML, like Gecko) Version/3.1.1 Mobile/5H11 Safari/525.20
Mozilla/5.0 (iPhone; U; CPU iPhone OS 2_2_1 like Mac OS X; en-us) AppleWebKit/525.18.1 (KHTML, like Gecko) Version/3.1.1 Mobile/5H11 Safari/525.20
Mozilla/5.0 (iPhone; U; CPU iPhone OS 2_0 like Mac OS X; en-us) AppleWebKit/525.18.1 (KHTML, like Gecko) Version/3.1.1 Mobile/5A347 Safari/525.20
Mozilla/5.0 (iPod; U; CPU iPhone OS 2_0 like Mac OS X; de-de) AppleWebKit/525.18.1 (KHTML, like Gecko) Version/3.1.1 Mobile/5A347 Safari/525.20
```

**Comment by jaydiablo** [comment]
Doing the same sort of analysis on Safari 4.x useragents, it appears that Safari 4.0 was only available for iOS 3.x and 4.x, specifically 3.0 -> 4.1.  It appears that Safari 5 was first released on iOS 4.2 or 4.3 (hard to say for sure, I have some UAs that overlap).

Wikipedia claims that Safari 4 was iOS 4.1:https://en.wikipedia.org/wiki/Safari_version_history#iOS  But it also has Safari 5.0 at iOS 4.2.1 (where's 4.2.0?)

I do see UAs for 4.1, but the build numbers don't match the stated iOS version, so I'm assuming these are spoofed (browscap doesn't have a 4.1 version defined anyhow).

I'm planning to remove the unnecessary iOS versions for Safari 4 (I'll keep 3.0 -> 4.2).  The iPad was released during this time (iOS 3.2 is the iPad only release), so the iPad patterns will remain.

**Comment by jaydiablo** [comment]
Ultimately this is what I've finished with after looking over a lot of different UAs and investigating the ones that ended up parsing different from `master` with the changes I'm making.

### Safari 3.0
* iOS 1.0 (iPhone/iPod only) It actually spans a few 1.x versions of iOS, but to detect those we have to look at the `Mobile/xxxx` token, which seems like unnecessary pattern additions for UAs that are around 10 years old and shouldn't really be seen in the wild any longer.

### Safari 3.1
* iOS 2.2 (iPhone/iPod only)
* iOS 2.1 (iPhone/iPod only)
* iOS 2.0 (iPhone/iPod only)

### Safari 4.0
* iOS 4.2 (some beta versions of the OS still on 4.0)
* iOS 4.1 (iPhone/iPod only)
* iOS 4.0 (iPhone/iPod only)
* iOS 3.2 (iPad only)
* iOS 3.1 (iPhone/iPod only)
* iOS 3.0 (iPhone/iPod only)

### Safari 5.0
* iOS 5.0 (some beta versions of the OS still on 5.0).
* iOS 4.3
* iOS 4.2

### Safari 5.1
* iOS 6.0 (some beta versions of the OS still on 5.1)
* iOS 5.1
* iOS 5.0

### Safari 6.0
* iOS 7.0 (some beta versions of the OS still on 6.0)
* iOS 6.1
* iOS 6.0

### Safari 7.0
* iOS 8.1 (This was the only one where I went +2 versions from the last major version match (8.0 and 8.1) as I saw a few useragents that looked legit otherwise (applewebkit version in the right range, mobile token matches iOS version, Safari build matches safari version number, no other signs of spoofing))
* iOS 8.0
* iOS 7.1
* iOS 7.0
* iOS 6.1 (I also went down one extra version for the same reasons as above)

### Safari 8.0
* iOS 9.0 (some beta versions of the OS still on 8.0)
* iOS 8.4
* iOS 8.3
* iOS 8.2
* iOS 8.1
* iOS 8.0

### Safari 9.0
* iOS 10.0 (some beta versions of the OS still on 9.0)
* iOS 9.3
* iOS 9.2
* iOS 9.1
* iOS 9.0

### Safari 10.0
* iOS 11.0 (some beta versions of the OS still on 10.0)
* iOS 10.3
* iOS 10.2
* iOS 10.1
* iOS 10.0

### Safari 11.0
* iOS 11.1 (currently in Beta, but adding it anyhow)
* iOS 11.0

Any other versions are being removed (10.1, 10.2, 10.3, etc...) as the only useragents I could find of those versions appear to be spoofed (mis-matched `Mobile/` and iOS version, wrong Safari build number, AppleWebKit build number out of range, etc...).

Few existing tests are affected by the changes, but a couple were. I've moved them to the invalid versions test file with descriptions as to why they're invalid.

PR coming shortly, just wrapping up some testing.
